src/Microsoft.Diagnostics.Monitoring.RestServer/DiagProcessFilter.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring.RestServer
+{
+    /// <summary>
+    /// There are effectively 3 sets of models for this:
+    /// The configuration model (ProcessFilterOptions), which does not contain criteria such as RuntimeId, and is always string based.
+    /// The REST call model (ProcessKey), which does not contain criteria such as Command Line, and is strongly typed.
+    /// This internal model, which contains all possible criteria and is string based. Only this model is used for the actual comparison.
+    /// </summary>
+    internal sealed class DiagProcessFilter
```

Comment:
Why not convert ProcessKey to ProcessFilterOptions and then make of this an implementation detail within the DiagnosicServices class?

Reply:
That's not convertible, since ProcessKey supports RuntimeId while ProcessFilterOptions doesn't.

Reply:
For sake of argument, if that's the only thing blocking this conversion, ProcessFilterOptions _could_ support RuntimeInstanceCookie and be documented, but in reality just won't be used via configuration very often. That would lower the complexity of this code quite a bit.

Reply:
- I don't think the current complexity is really all that high, it's just conversion from one object model to another.
- We no longer have coupling between ProcessKey and ProcessFilterOptions, we can change them independently.

Reply:
I don't think I agree however there's nothing wrong with it, so I won't press further at this time.